`src/python/pants/bsp/protocol_test.py::test_intellij_test` times out

Seen here and reported independently at https://github.com/pantsbuild/pants/pull/15649#issuecomment-1138560587 as well: https://github.com/pantsbuild/pants/runs/6611985344?check_suite_focus=true#step:13:256

Looks like:
```
...
15:33:25.50 [ERROR] Completed: Run Pytest - src/python/pants/bsp/protocol_test.py:tests failed (exit code 1).
============================= test session starts ==============================
collecting ... collected 2 items

src/python/pants/bsp/protocol_test.py::test_basic_bsp_protocol PASSED    [ 50%]
src/python/pants/bsp/protocol_test.py::test_intellij_test FAILED         [100%]

=================================== FAILURES ===================================
______________________________ test_intellij_test ______________________________

jvm_rule_runner = RuleRunner(build_root=/tmp/_BUILD_ROOTz0v0tc57)
jvm_lockfile = JVMLockfileFixture(lockfile=CoursierResolvedLockfile(entries=(CoursierLockfileEntry(coord=Coordinate(group='org.scala-...calatest_2.13', version='3.2.10', packaging='jar', classifier=None, strict=True), url=None, jar=None, excludes=None)]))

    @pytest.mark.jvm_lockfile(
        path="protocol-intellij.test.lock",
        requirements=[
            "org.scala-lang:scala-library:2.13.6",
            "org.scalatest:scalatest_2.13:3.2.10",
        ],
    )
    def test_intellij_test(jvm_rule_runner: RuleRunner, jvm_lockfile: JVMLockfileFixture) -> None:
        jvm_rule_runner.write_files(
            {
                "3rdparty/jvm/BUILD": jvm_lockfile.requirements_as_jvm_artifact_targets(),
                "3rdparty/jvm/default.lock": jvm_lockfile.serialized_lockfile,
                "BUILD": "scalatest_tests(name='main')",
                "Spec.scala": dedent(
                    """\
                    package org.pantsbuild.example
    
                    import org.scalatest.funspec.AnyFunSpec
    
                    import org.pantsbuild.example.lib.ExampleLib
    
                    class ExampleLibSpec extends AnyFunSpec {
                      describe("ExampleLib") {
                        it("should say hello") {
                          assert(ExampleLib.HELLO == "hello!")
                        }
                      }
                    }
                    """
                ),
                "lib/BUILD": "java_sources()",
                "lib/ExampleLib.java": dedent(
            assert build_targets.targets[0].language_ids == ("java", "scala")
    
            # buildTarget/sources
            sources = SourcesResult.from_json_dict(
                endpoint.request(
                    "buildTarget/sources", SourcesParams(target_ids).to_json_dict()
                ).result(timeout=15)
            )
            assert len(sources.items[0].sources) == 2
    
            # buildTarget/dependencySources - (NB: stubbed)
            _ = endpoint.request(
                "buildTarget/dependencySources", DependencySourcesParams(target_ids).to_json_dict()
            ).result(timeout=15)
    
            # buildTarget/resources - (NB: used only to index resources)
            _ = endpoint.request(
                "buildTarget/resources", ResourcesParams(target_ids).to_json_dict()
>           ).result(timeout=15)

src/python/pants/bsp/protocol_test.py:220: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Future at 0x7f0604072310 state=pending>, timeout = 15

    def result(self, timeout=None):
        """Return the result of the call that the future represents.
    
        Args:
            timeout: The number of seconds to wait for the result if the future
                isn't done. If None, then there is no limit on the wait time.
    
        Returns:
            The result of the call that the future represents.
    
        Raises:
            CancelledError: If the future was cancelled.
            TimeoutError: If the future didn't finish executing before the given
                timeout.
            Exception: If the call raised then that exception will be raised.
        """
        with self._condition:
            if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED]:
                raise CancelledError()
            elif self._state == FINISHED:
                return self.__get_result()
    
            self._condition.wait(timeout)
    
            if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED]:
                raise CancelledError()
            elif self._state == FINISHED:
                return self.__get_result()
            else:
>               raise TimeoutError()
E               concurrent.futures._base.TimeoutError

/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/concurrent/futures/_base.py:437: TimeoutError
- generated xml file: /tmp/process-executionQBlyNa/src.python.pants.bsp.protocol_test.py.tests.xml -


=========================== short test summary info ============================
FAILED src/python/pants/bsp/protocol_test.py::test_intellij_test - concurrent...
========================= 1 failed, 1 passed in 23.22s =========================
...
```

--- a/src/python/pants/bsp/protocol_test.py
+++ b/src/python/pants/bsp/protocol_test.py
@@ -215,9 +215,13 @@ def test_intellij_test(jvm_rule_runner: RuleRunner, jvm_lockfile: JVMLockfileFix
         ).result(timeout=15)
 
         # buildTarget/resources - (NB: used only to index resources)
+        # We set the timeout higher here due to CI flakes as documented in:
+        #   https://github.com/pantsbuild/pants/issues/15657
+        # This seems to paper over some slow interaction between this request
+        # and the LMDB store as noted in the ticket.
         _ = endpoint.request(
             "buildTarget/resources", ResourcesParams(target_ids).to_json_dict()
-        ).result(timeout=15)
+        ).result(timeout=45)
 
         # buildTarget/scalacOptions
         scalac_options = ScalacOptionsResult.from_json_dict(

--- a/src/python/pants/bsp/protocol_test.py
+++ b/src/python/pants/bsp/protocol_test.py
@@ -168,6 +168,12 @@ def test_intellij_test(jvm_rule_runner: RuleRunner, jvm_lockfile: JVMLockfileFix
 
     target_ids = (BuildTargetIdentifier("pants:default"),)
 
+    # We set a very high timeout here (was 15s) due to CI flakes as documented in:
+    #   https://github.com/pantsbuild/pants/issues/15657
+    # This seems to paper over some slow interaction between requests and the LMDB
+    # store as noted in the ticket.
+    timeout = 45
+
     with setup_bsp_server(
         jvm_rule_runner,
         notification_names={"build/taskStart", "build/taskProgress", "build/taskFinish"},
@@ -188,14 +194,14 @@ def test_intellij_test(jvm_rule_runner: RuleRunner, jvm_lockfile: JVMLockfileFix
                     "supportedScalaVersions": [],
                 },
             ).to_json_dict(),
-        ).result(timeout=15)
+        ).result(timeout=timeout)
 
         # build/initialized
         endpoint.notify("build/initialized")
 
         # workspace/buildTargets
         build_targets = WorkspaceBuildTargetsResult.from_json_dict(
-            endpoint.request("workspace/buildTargets").result(timeout=15)
+            endpoint.request("workspace/buildTargets").result(timeout=timeout)
         )
         assert len(build_targets.targets) == 1
         assert build_targets.targets[0].capabilities == BuildTargetCapabilities(can_compile=True)
@@ -205,29 +211,25 @@ def test_intellij_test(jvm_rule_runner: RuleRunner, jvm_lockfile: JVMLockfileFix
         sources = SourcesResult.from_json_dict(
             endpoint.request(
                 "buildTarget/sources", SourcesParams(target_ids).to_json_dict()
-            ).result(timeout=15)
+            ).result(timeout=timeout)
         )
         assert len(sources.items[0].sources) == 2
 
         # buildTarget/dependencySources - (NB: stubbed)
         _ = endpoint.request(
             "buildTarget/dependencySources", DependencySourcesParams(target_ids).to_json_dict()
-        ).result(timeout=15)
+        ).result(timeout=timeout)
 
         # buildTarget/resources - (NB: used only to index resources)
-        # We set the timeout higher here due to CI flakes as documented in:
-        #   https://github.com/pantsbuild/pants/issues/15657
-        # This seems to paper over some slow interaction between this request
-        # and the LMDB store as noted in the ticket.
         _ = endpoint.request(
             "buildTarget/resources", ResourcesParams(target_ids).to_json_dict()
-        ).result(timeout=45)
+        ).result(timeout=timeout)
 
         # buildTarget/scalacOptions
         scalac_options = ScalacOptionsResult.from_json_dict(
             endpoint.request(
                 "buildTarget/scalacOptions", ScalacOptionsParams(target_ids).to_json_dict()
-            ).result(timeout=15)
+            ).result(timeout=timeout)
         )
         assert scalac_options.items[0].classpath
         class_directory = Path(urlparse(scalac_options.items[0].class_directory).path)
@@ -237,7 +239,7 @@ def test_intellij_test(jvm_rule_runner: RuleRunner, jvm_lockfile: JVMLockfileFix
         compile_result = CompileResult.from_json_dict(
             endpoint.request(
                 "buildTarget/compile", CompileParams(target_ids).to_json_dict()
-            ).result(timeout=15)
+            ).result(timeout=timeout)
         )
         assert StatusCode(compile_result.status_code) == StatusCode.OK
         notifications.assert_received_unordered(

